input-profile.ts

```diff
@@ -0,0 +1,322 @@
+import {Component, Object3D} from '@wonderlandengine/api';
+import {HandTracking} from '@wonderlandengine/components';
+import {vec3, quat} from 'gl-matrix';
+import {property} from '@wonderlandengine/api/decorators.js';
+
+const _TempVec = vec3.create();
+const _TempQuat = quat.create();
+const minTemp = new Float32Array(3);
+const maxTemp = new Float32Array(3);
+const hands = ['left', 'right'];
+
+interface VisualResponse {
+    target: Object3D;
+    min: Object3D;
+    max: Object3D;
+    id: string;
+}
+
+export class InputProfile extends Component {
+    static TypeName = 'input-profile';
+
+    private _gamepadObjects: Record<string, any> = {};
+    private controllerModel: any;
+    private toFilter: Set<string> = new Set(['vr-mode-active-mode-switch']);
+    private defaultControllerComponents?: Component[];
+    private handedness!: string;
+    private url!: string;
+    private ProfileJSON: any = null;
+    private modelLoaded!: boolean;
+    private gamepad: any;
+
+    _buttons: VisualResponse[] = []; /* Any button, **EVEN** thumbstick click */
+    _axes: VisualResponse[] = [];
+
+    @property.object()
+    defaultController!: Object3D;
+
+    @property.enum(hands, 0)
+    handednessIndex: number = 0;
+
+    @property.string(
+        'https://cdn.jsdelivr.net/npm/@webxr-input-profiles/assets@latest/dist/profiles/'
+    )
+    path!: string;
+
+    @property.string()
+    customProfileFolder!: string;
+
+    @property.bool(false)
+    mapToDefaultController!: boolean;
+
+    @property.object()
+    trackedHand!: Object3D;
+
+    init() {
+        this._gamepadObjects = {};
+    }
+
+    start() {
+        this.controllerModel = null;
+        this.toFilter = new Set(['vr-mode-active-mode-switch']);
+        this.defaultControllerComponents = this.getComponents(this.defaultController);
+        this.handedness = hands[this.handednessIndex];
+
+        if (this.engine.xr?.session != null) {
```

Comment:
I still don't think this condition is needed. Can you try without? The `onXRSessionStart` is a `RetainEmitter`, what it does is exactly what you are doing here: If it's already been trigger, it will call your callback directly